subsys/bluetooth/host/smp.c

```diff
@@ -3059,6 +3057,13 @@ static int smp_send_pairing_req(struct bt_conn *conn)
 		return -EIO;
 	}
 
+	/* A higher security level is requested during the key distribution
+	 * phase, once pairing is complete a new pairing procedure will start.
+	 */
+	if (atomic_test_bit(smp->flags, SMP_FLAG_KEYS_DISTR)) {
+		return 0;
+	}
```

Comment:
SMP_FLAG_KEYS_DISTR is set from bt_smp_encrypt_change() so this depends on order of registering channels

since channels are registered via BT_L2CAP_CHANNEL_DEFINE macro order seems to depend on linker action

Reply:
I looked into the finished `zephyr.elf` and this is the order.
```
<att_fixed_chan>:
 <le_fixed_chan>:
<smp_fixed_chan>:
```
I'm pretty sure that the order is alphabetical, because the we seem to rely on this behavior for the static GATT service declararions, e.g:
```
<_1_gatt_svc>:
<_2_gap_svc>:

Reply:
fair enough :-)